pkg/infra/tower.go

```diff
@@ -124,16 +117,15 @@ func (t *Tower) hpDecay() {
 		agent.setHasEaten(false)
 		if agent.daysAtCritical >= t.healthInfo.MaxDayCritical {
 			t.Log("Killing agent", Fields{"agent": agent.id})
-			t.missingAgents[agent.floor] = append(t.missingAgents[agent.floor], agent.agentType)
-			delete(t.agents, agent.id) // maybe lock mutex?
-		} else {
-			agent.setHP(newHP)
+			newHP = 0
```

Comment:
why not remove it?

Reply:
Agents are deleted and replaced in the same place (`replaceAgents()`)